### PR TITLE
improve inferrabilities within TOML module

### PR DESCRIPTION
**Pull request by aviatesk**

### CODE REVIEW

File: base/toml_parser.jl [hunk 1]
```diff
@@ -1133,7 +1140,7 @@ function parse_string_continue(l::Parser, multiline::Bool, quoted::Bool)::Err{St
                     if !accept_n(l, n, isvalid_hex)
                         return ParserError(ErrInvalidUnicodeScalar)
                     end
-                    codepoint = parse_int(l, false, 16)
+                    codepoint = parse_int(l, false, 16)::Int
```
Comment (KristofferC) [inline base/toml_parser.jl:1143]:
```suggestion
                    codepoint = parse_int(l, false, 16)::Int64
```

I think

File: base/toml_parser.jl [hunk 2]
```diff
@@ -925,14 +925,16 @@ ok_end_value(c::Char) = iswhitespace(c) || c == '#' || c == EOF_CHAR || c == ']'
 accept_two(l, f::F) where {F} = accept_n(l, 2, f) || return(ParserError(ErrParsingDateTime))
 function parse_datetime(l)
     # Year has already been eaten when we reach here
-    year = parse_int(l, false)::Int64
+    year = parse_int(l, false)
```
Comment (KristofferC) [inline base/toml_parser.jl:928]:
```suggestion
    year = @try parse_int(l, false)
```

and remove the check below I think. Can do this in the other places as well.

File: base/toml_parser.jl [hunk 3]
```diff
@@ -976,9 +979,10 @@ function parse_datetime(l)
 end
 
 function try_return_datetime(p, year, month, day, h, m, s, ms)
-    if p.Dates !== nothing
+    Dates = p.Dates
```
Comment (KristofferC) [inline base/toml_parser.jl:982]: Nice catch

File: base/toml_parser.jl (hunk 2 above)
Comment (aviatesk) [reply base/toml_parser.jl:928]: done

**Review by KristofferC — APPROVED** [review]

**Pull request merged**